Lib/functools.py

```diff
@@ -273,6 +273,11 @@ def reduce(function, sequence, initial=_initial_missing):
 ### partial() argument application
 ################################################################################
 
+
+class Placeholder:
```

Comment:
Ideally, this should be implemented as a non-instantiable singleton (use `None` and `NoneType` as a model).

Reply:
Thank you - I was looking for a better solution for this.

```
try:
    from _functools import partial, Placeholder
except ImportError:
    pass
```

What do I use as a placeholder when C extensions is not loaded then?

Or can I put it outside of `functools`, say next to where `None` is defined (`object.c`)?

If creating a new sentinel, is it ok to create one with such a narrow use case?

PEP: https://peps.python.org/pep-0661/

Maybe it would be worth bringing the PEP back for reconsideration? Last time I brought this up was maybe a year ago, but it was more from seeking advice perspective, although was never fully happy about any of the solutions that I got recommended.

After chatting a bit about it and collecting advice, I ended up using `unittest.mock.sentinel`.

Lately (couple of weeks ago maybe), I wasn't happy about it anymore. So I took what I liked about `unittest`'s solution, added my own twist and ended up with:

https://gist.github.com/dg-pb/3dfa188af4a86b003b69078c0a73dbc7

,with which I am fairly happy about now.

It has attributes and one can define its methods too. Also, not depend on unit testing module for sentinel creation is a plus.

Maybe with some improvements, such as ability to create immortal instances from C (maybe even from python?) it could simplify things a bit.

Reply:
Or do I mimic `None` in "functools.c" and leave it as it is in "functools.py"?